### PR TITLE
fix(drive): uncommitted state if db transaction fails

### PR DESCRIPTION
**Pull request by shumkov**
## Issue being fixed or feature implemented

We had a sequence of errors on the mainnet started since block 32326. We got RocksDB's "transaction is busy" error because of a bug (#2309). Due to another bug in Tenderdash (https://github.com/dashpay/tenderdash/pull/966), validators just proceeded to the next block partially committing the state and updating the cache. Full nodes are stuck and proceeded after re-sync.

## What was done?

- Panic, if app hashes in memory and on disk, are different
- Panic if we aren't able to commit the Drive state to restart Drive and try this block from scratch

For the mainnet chain, we enable these fixes at the block when we consider the state is consistent.

## How Has This Been Tested?

Syncing testnet and mainnet

## Breaking Changes

None

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [x] I have made corresponding changes to the documentation if needed

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

## Summary by CodeRabbit

- **New Features**
	- Enhanced error handling and verification logic for transaction and proposal processing.
	- Improved consistency checks between in-memory and on-disk application states.

- **Bug Fixes**
	- Addressed specific inconsistencies related to block heights on the Dash network, particularly for block height 32326 and 32327.

- **Refactor**
	- Simplified the `PlatformConfig` structure by removing the `initial_protocol_version` field and related methods. 

- **Documentation**
	- Updated comments to clarify the rationale behind changes and temporary solutions for handling specific network conditions.

### CODE REVIEW

**Comment by shumkov** [comment]
@coderabbit review

File: packages/rs-drive-abci/src/abci/handler/info.rs [hunk 1]
```diff
@@ -21,28 +22,63 @@ where
...
+        .drive
+        .grove
+        .root_hash(
+            None,
+            &platform_state
+                .current_platform_version()?
+                .drive
+                .grove_version,
+        )
+        .unwrap()?;
```
[the remark this answers is not in the document]
Comment (shumkov) [reply packages/rs-drive-abci/src/abci/handler/info.rs]: This is unwrapping cost result. It won't panic with error, don't worry.

File: packages/rs-drive-abci/src/abci/handler/prepare_proposal.rs [hunk 2]
```diff
@@ -35,6 +36,44 @@ where
...
+                .grove_version,
+        )
+        .unwrap()?;
+
+    // We had a chain halt on mainnet on block 32326. Compaction happened
+    // and transaction.commit() returned an error. Due to a bug in tenderdash,
+    // validators just proceeded on next block without committing data but keeping
+    // updated cache. To keep consistency with mainnet chain we allow app hashes to be
+    // different for this block.
+    // TODO: verify that chain id is evo1
```
[the remark this answers is not in the document]
Comment (shumkov) [reply packages/rs-drive-abci/src/abci/handler/prepare_proposal.rs]: How can you assist?

[the remark this answers is not in the document]
Comment (shumkov) [reply packages/rs-drive-abci/src/abci/handler/prepare_proposal.rs]: `chain_id` is located in `config.abci.chain_id` but thank you anyways. It's nice.

**Review by QuantumExplorer — DISMISSED** [review]

**Review by QuantumExplorer — APPROVED** [review]

**Pull request merged**